packages/cli/templates/react/src/components/FieldPlugin.tsx

```diff
@@ -3,7 +3,26 @@ import { useFieldPlugin } from '@storyblok/field-plugin/react'
 
 const FieldPlugin: FunctionComponent = () => {
   const plugin = useFieldPlugin({
-    parseContent: (content: unknown) => content,
+    /*
+    The `validateContent` parameter is optional. It allows you to
+      - validate the content
+      - make changes before sending it to the Storyblok Visual Editor
+      - provide type-safety
+
+    // For example,
+    validateContent: (content: unknown) => {
+      if (typeof content === 'string') {
+        return {
+          content,
+        }
+      } else {
+        return {
+          content,
```

Comment:
should we pass content anyway even if there is error ?

Reply:
I find this topic a bit difficult.

Firstly, the field plugin library should always send the content (the return value of `validateContent`) to the Visual Editor whenever `setContent()` is called. Whether to accept it or not is a decision on the Visual Editor side. It's similar to a client-server relationship.

Now, let's discuss the second part: should users return content as well or only an error from the `validateContent` function when it's invalid?

If the Visual Editor is capable of preventing saving based on the validation result (which is not currently the case), then we can encourage developers to send the content even when it's invalid. This approach makes sense because the Visual Editor has the data but won't save it due to the error message.

However, in the current situation, the Visual Editor is unable to do that. If someone really wants to prevent saving invalid data, they can simply return `{ content: undefined, error: "..." }`. Nevertheless, this situation should not last for long. We should update the Visual Editor to address this limitation.

If they return "undefined" for content and we later update the Visual Editor, we will need to communicate again so that they can update their field plugins. While this may not be a strong argument, I believe that "returning content all the time" could be a suitable default in most cases.

What do you think?

Reply:
I understand your point where you explain why content should be sent to the editor all the time. 

For me now the confusion happens because you mention that the Visual Editor should decide whether to accept the setContent or not. But in our implementation actually the library decides. It is not a problem but from conceptual point of view it might be unclear, and we might need to add this info into our docs. 

Lets makes this change and we will see how we feel about it once we start testing it. So it's an ok from me :)

Reply:
@BibiSebi 

> But in our implementation actually the library decides

Could you elaborate on this part? 

If you look at this part, 
https://github.com/storyblok/field-plugin/blob/c0e0b553e3ce6350f73c8e7ffe9b39b4b7b9d440/packages/field-plugin/src/createFieldPlugin/createPluginActions/partialPluginStateFromStateChangeMessage.ts#L23-L34

no matter if it's valid or not, the library still sends the content to the Visual Editor.

So I'm not sure if I understand you correctly about "the library decides".

Reply:
In the future (but also now), the library does the validation, then sends the object including content and error property to storyblok. So storyblok per se does not validate really it just checks if an object including error is incoming from the plugin and does whatever it needs to (like preventing from hitting save and showing an error message - in future).

As a developer who uses field plugin, I probably would think that the validation is done inside the storyfront. But at this point it does not matter as much, so we can merge it. I was only mentioning it because I am wondering if there can be any problems related to doing validation inside the SDK. So just food for thought, but not blocking.

Reply:
Oh I see what you mean. Yeah, ideally validating something before saving it into database should happen on the server side too. Now it's not on the server side. It's not even on the Storyblok side, but it's on the field plugin side.